packages/yoga/src/Dialog/web/utils.js

```diff
@@ -0,0 +1,5 @@
+export function focusOnFirstProgrammaticFocusableElement(container) {
+  const element = container.querySelector('[tabindex="-1"]');
+
+  element?.focus();
+}
```

Comment:
Why are you focusing on items that have been removed from navigation?

If an element is not naturally focusable, we use tabindex="0"

As we are using react, I believe that useRef is more suitable for focusing on elements.

Use .focus() only when necessary.
Apply tabindex="0" to non-focusable but relevant elements.
Ensure focus cycling within components such as modals.
Add visual feedback on focus when necessary.
Work with accessibility attributes to guide assistive technologies.
These practices ensure an inclusive and consistent experience for all users.

Reply:
This function is called by the `onActivation` prop from `FocusLock`, in this case the container will be the `FocusLock` itself, according to Figma's specs in most scenarios the initial focus will be on the title.

These adjustments were made following the w3c practices guide, the only attribute I didn't add was `aria-labelledby` because of the `Drawer` scenario, but since the initial focus will always be on the title I don't think this will be a problem.
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/

The use of `tabindex=-1` is for scenarios where a non-interactive element is focused programmatically.
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/